Makefile

```diff
@@ -21,14 +21,10 @@ download_prebuilt:
 .PHONY: install
 install: printer.arm
 	ssh-add
-	ssh root@$(host) systemctl stop printer
+	ssh root@$(host) systemctl stop printer 2>&1 >/dev/null
 	scp printer.arm root@$(host):
 	scp printer.service root@$(host):/etc/systemd/system
-	ssh root@$(host) <<- ENDSSH
-		systemctl daemon-reload
-		systemctl enable printer
-		systemctl restart printer
-	ENDSSH
+	ssh root@$(host) 'systemctl daemon-reload && systemctl enable printer && systemctl restart printer'
```

Comment:
Please leave the Heredoc as is.  The pseudo-tty message is not an error.

Reply:
I don't believe it worked without these changes, but I'll double check and report.

Reply:
If you simply change all of this to

    ssh root@$(host) systemctl stop printer || true

does that fix the issue?

Reply:
Make 3.81 (shipped on macOS) doesn't seem to support heredoc syntax. After the change proposed by @Evidlo, it works.

I also had to remove the ssh-add line since on macOS it is common to load all the SSH keys using passwords from the Keychain with `ssh-add -A`. After that, `ssh-add` will (unnecessarily) ask for passwords for all the SSH keys.

Both things are really minor, thanks for the awesome tool @Gozala!

Reply:
For the first issue of printer service not loaded/available, the fix proposed by @Evidlo works for me. 

The second one, Pseudo-terminal will not be allocated - I tried forcing tty with -tt option, it closes ssh connection - trading an error for a warning :) But the fix proposed by @Gozala is much cleaner (using macOS 11). Thanks @Gozala.

Thanks for amazing utility @Evidlo

Reply:
Can someone do a test for me?  I'd like to get to the bottom of this, as heredoc supported is determined by the shell (zsh on OSX).

What does `make test` do for this makefile?

```
.ONESHELL:

.PHONY: test
test:
	cat <<- ENDCAT
		hello world
		foobar123
		more text
	ENDCAT
```

If that doesn't work, does adding `SHELL=/bin/bash` to the top fix anything?

Reply:
Here is the output for above code - 
```
cat <<- ENDCAT
hello world
make: hello: No such file or directory
make: *** [test] Error 1
```

Running it with -i (ignore errors) 
```
cat <<- ENDCAT
hello world
make: hello: No such file or directory
make: [test] Error 1 (ignored)
foobar123
make: foobar123: No such file or directory
make: [test] Error 1 (ignored)
more text
text: No such file or directory
make: [test] Error 1 (ignored)
ENDCAT
make: ENDCAT: No such file or directory
make: [test] Error 1 (ignored)
```

Additional Info  - 
Make version 3.81  built for i386-apple-darwin11.3.0
zsh 5.8 (x86_64-apple-darwin20.1.0).  -- Current Shell 
bash version 3.2.57(1)-release (x86_64-apple-darwin20)

Adding shell directive in the makefile did not help fix the issue.  

Let me know if you need more info

Reply:
If you just put this in `test.sh` and try to run it (in zsh), do you get the same thing?

```
cat <<- ENDCAT
		hello world
		foobar123
		more text
ENDCAT
```

I've tested with zsh on Linux and both this and the Makefile work fine.

Reply:
OK.  The issue is that Make 3.81 doesn't support the ONESHELL.

Also 3.81 is over 10 years old now.  Seriously Apple?

Reply:
The script (test.sh) worked fine in zsh and bash